### PR TITLE
Add data to API

### PR DESCRIPTION
**Pull request by Infernia829**
I added a list of connected players to the API, its confirmed working on a local server.

Attached below is a picture of the resulting .json.

confirmed_working

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -9,6 +22,18 @@ if table.indexof(minetest.get_dir_list(worldpath, true), "api") == -1 then
...
+	local player_names = {}
+	local players = minetest.get_connected_players()
+
+	for _, player in ipairs(players) do
+    	local name = player:get_player_name()
+    	table.insert(player_names, name)
+	end
+
+	return minetest.write_json(player_names), table.getn(player_names)
+end
```
Comment (LoneWolfHT) [inline init.lua]:
```suggestion
local function get_player_list()
	local player_names = {}

	for _, player in ipairs(minetest.get_connected_players()) do
    	table.insert(player_names, player:get_player_name())
	end

	return player_names, #player_names
end
```

File: init.lua [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+local timer = 0
+
+-- data cache --
+
+local name_mode = ''
+local matches_mode = ''
+local matches_played_mode = ''
+local name_map = ''
+local technical_name_map = ''
+local start_time_map = ''
```
Comment (LoneWolfHT) [inline init.lua]:
```suggestion
local cache = {}
```

File: init.lua [hunk 3]
```diff
@@ -18,16 +43,59 @@ local function update(data)
...
+
+	local player_names, player_count = get_player_list()
+
+	name_map = ctf_map.current_map.name
+	technical_name_map = ctf_map.current_map.dirname
+	start_time_map = os.time() -- Can be converted to local date here https://www.unixtimestamp.com/
+	name_mode = ctf_modebase.current_mode
+	matches_mode = ctf_modebase.current_mode_matches
+	matches_played_mode = ctf_modebase.current_mode_matches_played
+
```
Comment (LoneWolfHT) [inline init.lua]:
```suggestion
```

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]
My suggestions will get you started on the proper implementation

File: init.lua (hunk 2 above)
Comment (Infernia829) [reply init.lua]: I thought there was a better way, just plumb slipped my mind.

File: init.lua (hunk 3 above)
Comment (LoneWolfHT) [reply init.lua]: Basically just revert the changes you made when adding these non-playerlist variables

**Comment by Infernia829** [comment]
Changes pushed, confirmed working: 
image

File: init.lua [hunk 4]
```diff
@@ -18,16 +34,56 @@ local function update(data)
...
+			current_map = {
+				name = cache[3],
+				technical_name = cache[4],
+				start_time = cache[5], 
+			},
+			player_info = {
+				players = player_names,
+				count = player_count
+			},
+		})
```
Comment (LoneWolfHT) [inline init.lua]:
```suggestion
        cache.player_info = {
             players = player_names,
             count = player_count
        }
        update(cache)
```

**Comment by LoneWolfHT** [comment]
For some reason my fourth suggestion didn't appear, see my latest one on how to use the `cache` var

**Comment by Infernia829** [comment]
Made changes based of of @LoneWolfHT's suggestions, and stopped using globalstep() in favor of joinplayer/leaveplayer.

Confirmed working:

image

**Review by LoneWolfHT — APPROVED** [review]

**Pull request merged**